scIB/integration.py

```diff
@@ -174,6 +174,69 @@ def runScvi(adata, batch, hvg=None):
     adata.obsm['X_emb'] = latent
 
     return adata
+def runScanvi(adata, batch, labels, hvg=None):
+    # Use non-normalized (count) data for scvi!
+    # Expects data only on HVGs
+    
+    scIB.utils.checkSanity(adata, batch, hvg)
+
+    # Check for counts data layer
+    if 'counts' not in adata.layers:
+        raise TypeError('Adata does not contain a `counts` layer in `adata.layers[`counts`]`')
+
+    from scvi.models import VAE, SCANVI
+    from scvi.inference import AlternateSemiSupervisedTrainer, SemiSupervisedTrainer
+    from sklearn.preprocessing import LabelEncoder
+    from scvi.dataset import AnnDatasetFromAnnData
+
+    # Defaults from SCVI github tutorials scanpy_pbmc3k and harmonization
+    n_epochs=np.min([round((20000/adata.n_obs)*400), 400])
+    n_latent=30
+    n_hidden=128
+    n_layers=2
+    
+    net_adata = adata.copy()
+    net_adata.X = adata.layers['counts']
+    del net_adata.layers['counts']
+    # Ensure that the raw counts are not accidentally used
+    del net_adata.raw # Note that this only works from anndata 0.7
+
+    # Define batch indices
+    le = LabelEncoder()
+    net_adata.obs['batch_indices'] = le.fit_transform(net_adata.obs[batch].values)
+    net_adata.obs['labels'] = le.fit_transform(net_adata.obs[labels].values)
+
+    net_adata_out = AnnDatasetFromAnnData(net_adata)
+    net_adata_out.labels = net_adata.obs['labels']
+
```

Comment:
Usually, we learn scVI first, and then on the initialized weights of scVI run the scANVI training. I need Pierre to verify this.

Reply:
I see this in the tutorial now, yes... this should be added. Thanks for pointing it out. I guess that makes both methods intertwined (as they are anyway), but that's fine if this is the canonical way to run scANVI.

Reply:
yes, we have been experimenting with training procedures for quite a while. This is the most stable way of running the method.